### PR TITLE
fix nxos_vpc_interface peer_link idempotence

### LINKED ISSUE #35736: nxos_vpc_interface peer_link is not idempotent
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

nxos_vpc_Interface
##### ANSIBLE VERSION

```
ansible 2.5.0 (devel 788010d0f0) last updated 2018/01/08 11:49:21 (GMT -400)
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /root/agents-ci/ansible/lib/ansible
  executable location = /root/agents-ci/ansible/bin/ansible
  python version = 2.7.6 (default, Oct 26 2016, 20:30:19) [GCC 4.8.4]
```

##### OS / ENVIRONMENT

Ansible Server : Ubuntu 14.04
Device: N9K running 7.0(3)I5(2)

##### SUMMARY

If we set peer_link to False, it is not idempotent. Setting it to True is ok.

Note: I will open a PR shortly to address the above issue

### PR DESCRIPTION
**Pull request by saichint**
##### SUMMARY

Fixes #35736

##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME

nxos_vpc_interface

##### ANSIBLE VERSION

```
ansible 2.5.0 (devel 788010d0f0) last updated 2018/01/08 11:49:21 (GMT -400)
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /root/agents-ci/ansible/lib/ansible
  executable location = /root/agents-ci/ansible/bin/ansible
  python version = 2.7.6 (default, Oct 26 2016, 20:30:19) [GCC 4.8.4]
```

##### ADDITIONAL INFORMATION
* This PR fixes issues in #35736
* Integration test cases are enhanced for nxos_vpc_interface module to include all parameters
* Tested on all possible platforms and versions.

```

```

### CODE REVIEW

File: lib/ansible/modules/network/nxos/nxos_vpc_interface.py [hunk 1]
```diff
@@ -194,11 +194,14 @@ def get_commands_to_config_vpc_interface(portchannel, delta, config_value, exist
     commands = []
 
     if delta.get('peer-link') is False and existing.get('peer-link') is True:
-        command = 'no vpc peer-link'
         commands.append('no vpc peer-link')
         commands.insert(0, 'interface port-channel{0}'.format(portchannel))
 
-    elif delta.get('peer-link') or not existing.get('vpc'):
+    elif delta.get('peer-link') and existing.get('peer-link') is not True:
```
Comment (trishnaguha) [inline lib/ansible/modules/network/nxos/nxos_vpc_interface.py]: Please modify this to `elif delta.get('peer-link') and not existing.get('peer-link'):`

File: lib/ansible/modules/network/nxos/nxos_vpc_interface.py [hunk 2]
```diff
@@ -194,11 +194,14 @@ def get_commands_to_config_vpc_interface(portchannel, delta, config_value, exist
     commands = []
 
     if delta.get('peer-link') is False and existing.get('peer-link') is True:
```
Comment (trishnaguha) [inline lib/ansible/modules/network/nxos/nxos_vpc_interface.py]: Modify this one as well? `if not delta.get('peer-link') and existing.get('peer-link'):`

**Review by trishnaguha — CHANGES_REQUESTED** [review]

**Comment by trishnaguha** [comment]
@saichint please push the requested change and we are good to merge after that.

**Comment by saichint** [comment]
Made changes and pushed

**Review by trishnaguha — APPROVED** [review]

**Pull request merged**